mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl

```diff
@@ -4157,6 +4157,65 @@ resource "google_container_node_pool" "np" {
 `, clusterName, networkName, subnetworkName, np)
 }
 
+func TestAccContainerNodePool_withLocalSsdEncryptionMode(t *testing.T) {
+	t.Parallel()
+
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", acctest.RandString(t, 10))
+	np := fmt.Sprintf("tf-test-cluster-nodepool-%s", acctest.RandString(t, 10))
+	networkName := acctest.BootstrapSharedTestNetwork(t, "gke-cluster")
+	subnetworkName := acctest.BootstrapSubnet(t, "gke-cluster", networkName)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:	  func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerNodePool_withLocalSsdEncryptionMode(clusterName, np, networkName, subnetworkName, "EPHEMERAL_KEY_ENCRYPTION"),
```

Comment:
Can we add a second config step with an updated value, since it's mutable? https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test

Only this test is fine- GKE cluster shares code so it'd be redundant.

Reply:
side note: if you want to still test the update path in a test, instead of checking for no deletion in the plancheck, you can just not check whether anything's deleted, or you can do something like this instead:
```
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("xxx.yyy", plancheck.ResourceActionDestroyBeforeCreate),
					},
				},
```